CoreUtilities/src/au/gov/asd/tac/constellation/utilities/rest/RestClient.java

```diff
@@ -160,40 +266,133 @@ public void get(final String url, final Map<String, String> params) throws IOExc
+     */
+    public void get(final String url, final List<Tuple<String, String>> params) throws IOException {
+        beforeGet(url, params);
+
+        HttpsURLConnection connection = null;
+        try {
+            connection = makeGetConnection(url, params);
+
+            responseCode = connection.getResponseCode();
+            responseMessage = connection.getResponseMessage();
+            headerFields = connection.getHeaderFields();
+
+            bytes = null;
+            if (Response.isCodeSuccess(responseCode)) {
+                bytes = getBody(connection, responseCode);
+            }
+        } finally {
+            if (connection != null) {
+                connection.disconnect();
+            }
+        }
+        afterGet(url, params);
+    }
 
     /**
-     * Will be run after the GET connection has been disconnected
+     * Will be run after the GET connection has been disconnected - designed to
+     * be overridden as required.
+     * Deprecated - use afterGet(final String url, final List<Tuple<String, String>> params)
+     * instead.
      *
      * @param url The URL to request.
      * @param params Query parameters.
+     * @deprecated
      */
+    @Deprecated
     public void afterGet(final String url, final Map<String, String> params) {
         // DO NOTHING
     }
 
-    public abstract HttpsURLConnection makePostConnection(final String url, final Map<String, String> params) throws IOException;
+    /**
+     * Will be run after the GET connection has been disconnected - designed to
+     * be overridden as required.
+     *
+     * @param url The URL to request.
+     * @param params Query parameters.
+     */
+    public void afterGet(final String url, final List<Tuple<String, String>> params) {
+        // DO NOTHING
+    }
+    
+    /**
+     * Manage the creation of a HTTP POST connection.
+     * Deprecated - use makePostConnection(final String url, final List<Tuple<String, String>> params)
+     * instead.
+     * 
+     * @param url The URL to base the POST connection on.
+     * @param params Any parameters to add to the URL
+     * @return HttpsURLConnection object corresponding to the created connection
+     * @throws IOException 
+     * @deprecated
+     */
+    @Deprecated
+    public HttpsURLConnection makePostConnection(final String url, final Map<String, String> params) throws IOException {
+        throw new UnsupportedOperationException("This method is obsolete");
+    }
+    
+    /**
+     * Manage the creation of a HTTP POST connection.
+     * 
+     * @param url The URL to base the POST connection on.
+     * @param params Any parameters to add to the URL
+     * @return HttpsURLConnection object corresponding to the created connection
+     * @throws IOException
+     */
+    public HttpsURLConnection makePostConnection(final String url, final List<Tuple<String, String>> params) throws IOException {
+        throw new UnsupportedOperationException("This method must be overridden");
```

Comment:
similar

Reply:
As above

Reply:
Following our chat, Ive removed all the deprecated methods, which enables me to again make these methods abstract.